api/variables/aggregate_types.go

```diff
@@ -1,18 +1,19 @@
 // Copyright 2024 Nutanix. All rights reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-package clusterconfig
+package variables
 
 import carenv1 "github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/api/v1alpha1"
 
-// ClusterConfig is a type to be used internally to simplify the handling of cluster configurations for different
+// The types here are to be used internally to simplify the handling of cluster configurations for different
 // providers. It is not meant to be used as a CRD.
 // By including all the possible configurations for all the providers, we can easily switch between providers in code
 // without type assertions/switches and avoids passing around `interface{}` or `any` types.
 // Every provider-specific cluster config variable will successfully unmarshal to this type and so it is safe to use
 // this internally when a handler provides functionality for multiple providers but exhibits different behaviour per
 // provider.
-type ClusterConfig struct {
+
+type ClusterConfigSpec struct {
```

Comment:
Renamed the type here to better match the actual API type, dropping the infra provider prefix. https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/blob/7ccc23bf712cf2e2f21fa26cf83ff2f277c669e7/api/v1alpha1/clusterconfig_types.go#L68